discovery_engine_core/web-api/src/handlers.rs

```diff
@@ -45,7 +108,13 @@ pub(crate) async fn handle_user_interaction(
     body: InteractionRequestBody,
     db: Db,
 ) -> Result<impl warp::Reply, Rejection> {
-    if let Some(document) = db.documents_by_id.get(&body.document_id) {
+    let documents = db
+        .elastic_state
+        .get_documents_by_ids(vec![body.document_id])
```

Comment:
:art: we probably should adjust the function signature of `get_documents_by_ids()` to take a slice of ids instead of a vec to avoid unnecessary allocations.